python/pyarrow/tests/test_csv.py

```diff
@@ -306,6 +312,58 @@ def test_header_column_names(self):
             "y": ["kl", "op"],
             }
 
+    def test_include_columns(self):
```

Comment:
@nealrichardson You may want to look at the tests below and validate them.

Reply:
LGTM aside from my one question about order